### PR TITLE
[helm]: forward agent providers configuration for fleet mode

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
## What does this PR do?

This PR forwards providers configuration for managed agents (fleet mode)

## Why is it important?

Because users need to be able to disable/enabled leader election or/and kubernetes providers for managed agents

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Disruptive User Impact

N/A

## How to test this PR locally

```
mage helm:renderExamples
mage integration:kubernetesMatrix
```

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/5558

### CODE REVIEW

**Comment by michalpristas** [comment]
please don't forget to add proper `Team` label

**Review by swiatekm — APPROVED** [review]
LGTM, but I think we should have an example for fleet mode as well. Can be added in a follow up.

File: deploy/helm/elastic-agent/examples/fleet-managed/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+     --set agent.fleet.token=$FLEET_TOKEN \
+     --set agent.fleet.preset=perNode
+     -n kube-system
+```
+
+## Validate:
+
+1. [Optional] Install kube-state metrics if you want to see the KSM related metrics `kubectl apply -k https://github.com/kubernetes/kube-state-metrics`.
+2. Install Kubernetes integration to the agent policy that you created in Fleet. If you didn't install kube-state metrics from above, make sure to disable them in the integration.
+3. The Kibana `kubernetes`-related dashboards should start showing up the respective info.
```
Comment (swiatekm) [inline deploy/helm/elastic-agent/examples/fleet-managed/README.md:23]:
```suggestion
3. The Kibana `kubernetes`-related dashboards should start showing the respective info.
```

**Review by swiatekm — APPROVED** [review]
:+1:

**Pull request merged**